source/states/PlayerState.hx

```diff
@@ -0,0 +1,71 @@
+package states;
+
+import flixel.FlxG;
+import flixel.FlxObject;
+import flixel.input.keyboard.FlxKey;
+
+/**
+ * Any player related states should extend this one as it provides several
+ * convenience methods that will surely be needed
+ *
+ * @author Garren Ijames
+ */
+class PlayerState extends BaseState
+{
+  private var isRunning:Bool;
+  private var onGround:Bool;
+
+  public function new()
+  {
+    isRunning = false;
+
+    // This initial value might change depending upon how levels begin
+    onGround  = false;
+
+    super();
+  }
+
+  /**
+   * Convenience method for polling for horizontal movement as
+   * most of the player states need to take it into account
+   *
+   * @return scalar value of the players next horizontal move
+   */
+  private function pollForHorizontalMove():Int
```

Comment:
Two things: 
 1. you might want to have a function to poll for other sorts of inputs, like jumping
 2. I feel like it may be more appropriate to include this stuff in the Player class, since I feel that the finite state machine is focused on interpreting inputs rather than getting them in the first place.